### ISSUE TITLE
byte and int should be type aliases, not distinct types

### ISSUE THREAD

**Issue by ntrel** [vlang/v#983]
Go supports both distinct types (like C typedef but without implicit conversion) and true type aliases:
```
type New = Old
```
https://golang.org/ref/spec#Alias_declarations

* In Go, `byte` is a type *alias* for `uint8`, not a distinct type. So a `byte` can be assigned a `uint8` and vice versa, a `byte` can be passed to a `uint8` function parameter etc. Currently V does not allow these implicit conversions.
* In V, `int` should be an alias for `i32`, not a distinct type. ATM `i32` does not convert to `int` and vice versa.
* In Go, `rune` is also an alias, but I really think it should be a distinct type, as it is in V now. Then method `(r rune).str()` interprets `r` as a UTF-8 code point, and `(i int).str()` interprets `i` as a number (see also #216).

A possible way to simplify V would be to remove support for `type New Old` type definitions. I think they are almost the same as making a `struct New` with one field, except for `New(old)` vs `New{old}` syntax [assuming embedding and operator overloading make the struct act like `Old`].

**Comment by cytown** [comment]
According to this, the float should be a alias to f32?
No double right now, should be alias to f64?

**Comment by medvednikov** [comment]
There's no `float` or `double`.

**Comment by stappersg** [comment]
FWIW  I fail to see why this issue is currently labelled as "bug".
IMNSHO is this issue just a "wish".

**Plus**  an `int` are four `byte`s ...

**Comment by ntrel** [comment]
@stappersg Can you explain how byte is semantically different from i8, and int from i32? If not why are they different types? Each pair should refer to the same underlying type from the compilers point of view. Having to cast between them is just noise for no reason.

**Comment by stappersg** [comment]
> @stappersg Can you explain how byte is semantically different from i8, and int from i32? If not why are they different types? Each pair should refer to the same underlying type from the compilers point of view. Having to cast between them is just noise for no reason.

Dear [Sea Lion](https://en.wikipedia.org/wiki/Sealioning),

It is up to you to accept that `v` is not `go`.
Start with replacing the *bug label* on this issue into a *feature request label*.

Regards
Geert Stappers

P.S.
Thanks for the issues that you reported.
Please, pretty please, stop with default labelling them as *bug*.

**Comment by ntrel** [comment]
@stappersg The basic types docs say
```byte // alias for u8  
int  // alias for i32```

This is not the case, they are separate types, hence this is actually a bug.

**Comment by stappersg** [comment]
Okay, filtered the opening description of this issue, it boils down to

>In V, `int` should be an alias for `i32`, not a distinct type. ATM `i32` does not convert to `int` and vice versa.

**Comment by gslicer** [comment]
Based on the most recent change, it seems it was decided not to support aliases at all, instead to rename the type` i32` => `int` and `u8` => `byte` (although I would prefer to keep them consitent with the other types too), previously they were seperate types which was wrong and thus the bug label was correct based on the official "docs" (tutorial)

**Comment by medvednikov** [comment]
For simplicity and consistency I decided to remove primitive aliases. `i32` and `u8` might be back and replace `int` and `byte` in the future.

Aliases will still be supported.

**Comment by eduardolat** [comment]
i like the idea of not having `int` and `byte` types and use `i32` and `u8` directly, it's way more consistent with all other types

**Comment by JalonSolov** [comment]
So far, byte is gone...
```
x.v:1:6: error: byte is deprecated, use u8 instead
    1 | a := byte(0)
      |      ~~~~~~~
```
but `int` is here to stay... only it will be changing, too.  It will be `i32` on 32-bit systems, and `i64` on 64-bit systems.

**Comment by eduardolat** [comment]
The behavior of `int` that you describe makes more sense than its current behavior. Right now, it breaks the consistency of the type names

**Comment by stappersg** [comment]
On Fri, Jan 09, 2026 at 06:06:29PM -0800, Luis Eduardo wrote:
> Right now, it breaks the consistency of the type names

Patches welcome

 
Groeten
Geert Stappers
-- 
Silence is hard to parse

**Comment by JalonSolov** [comment]
The new `int` behavior is already implemented, but it is not the default.  You have to add `-new-int` on the V command line to use it.

Testing it with your own code, and opening GitHub issues will help resolve any problems with the code, and help make it the default sooner.
